File: src/main/java/com/leek/wars/bot/entities/responses/Response.java
package com.leek.wars.bot.entities.responses;

import com.leek.wars.bot.entities.Error;

public class Response {

	private Boolean success;
	private Error error;

	public Boolean isSuccess() {
		return success;
	}
	public void setSuccess(Boolean success) {
		this.success = success;
	}
	public Error getError() {
		return error;
	}
	public void setError(Error error) {
		this.error = error;
	}
	public void setError(String error) {
		this.error = Error.getByErrorStr(error);
	}
	
	
	@Override
	public String toString() {
		return "Response [success=" + success + ", error=" + error + "]";
	}
}

File: src/main/kotlin/com/leek/wars/bot/entities/responses/Response.kt
package com.leek.wars.bot.entities.responses

import com.leek.wars.bot.entities.Error

open class Response {
    var isSuccess: Boolean? = null
    var error: Error? = null

    fun setError(error: String) {
        this.error = Error.Companion.getByErrorStr(error)
    }

    override fun toString(): String {
        return "Response [success=" + isSuccess + ", error=" + error + "]"
    }
}